sklearn/decomposition/kernel_pca.py

```diff
@@ -295,8 +319,17 @@ def transform(self, X):
         """
         check_is_fitted(self, 'X_fit_')
 
+        # Compute centered gram matrix between X and training data X_fit_
         K = self._centerer.transform(self._get_kernel(X, self.X_fit_))
-        return np.dot(K, self.alphas_ / np.sqrt(self.lambdas_))
+
+        # scale eigenvectors
+        scaled_alphas = self.alphas_ / np.sqrt(self.lambdas_)
```

Comment:
I think we should only divide by the non-zero lambdas to avoid the zero division warning

Reply:
Granted. I will push this right away